integration-tests/tests/cases/solana.rs

```diff
@@ -40,3 +40,45 @@ async fn test_solana_signature_basic() -> anyhow::Result<()> {
         anyhow::bail!("signature verification failed");
     }
 }
+
+// Concurrent variant: spawn many sign requests at once against a very small
+// presignature stockpile and assert we make forward progress (no livelock).
+#[test(tokio::test)]
+async fn test_solana_stockpile_depletion() -> anyhow::Result<()> {
```

Comment:
Does this test actually test the case that was problematic? I would have expected some test that has Ps with a partial participant set, combined with offline nodes that are listed as participant.

Reply:
it does but it's not that explicit about it. I added tests in components instead and got rid of this test to showcase it further. The idea is to have double the amount of sign requests than we have a stockpile. The stockpile should build up which means that our pending sign requests should complete and not contend with each other. So for cases where the proposer isn't agreed upon or when we burn too many presignatures, this should catch it.

Reply:
Thanks! The new tests are much better!

But still, does it actually test the case that was problematic? In other words, would the tests have failed before this PR? Sorry if I'm missing something but I didn't find how the test triggers the condition `intersect_vec(&[&taken.artifact.participants, &stable]).len() < ctx.threshold` which seems necessary for the problem to occur.